lib/portlayer/exec/container.go

```diff
@@ -119,6 +127,27 @@ func (c *Container) NewHandle() *Handle {
 	return newHandle(c)
 }
 
+// Refresh calls the propery collector to get config and runtime info
+func (c *Container) Refresh() error {
+	var o mo.VirtualMachine
+
+	// make sure we have vm
+	if c.vm == nil {
+		return fmt.Errorf("There is no backing VirtualMachine")
+	}
+	ctx := context.TODO()
+
+	if err := c.vm.Properties(ctx, c.vm.Reference(), []string{"config", "runtime"}, &o); err != nil {
+		return err
+	}
+
+	c.Config = o.Config
+	c.Runtime = &o.Runtime
+
```

Comment:
I understand that things can change but I'm not sure how this new code is changing what we return in those cases? Whoever was handling those should be still doing it. 

We are calling this 2 get two things, the runtime.PowerState so that we can modify ExtraConfig and later we will start to get ChangeVersion attribute for making API calls and ensuring nothing changed since we last checked.